Makefile

```diff
@@ -287,3 +304,15 @@ kill-plugin:
 		echo "Killing plugin pid $$PID"; \
 		kill -9 $$PID; \
 	done; \
+
+.PHONY: validate-go-version
+validate-go-version: ## Validates the installed version of go against Mattermost's minimum requirement.
```

Comment:
Why is this needed for this PR? I'm not against the change in general, but IMO it shouldn't be part of this PR.

Reply:
I was lifting worthy go pieces from mattermost-server, this one came with it, why not?

I'll make a separate PR for the template, but I'd like this one to merge very soon since the merge conflicts are really annoying.

Reply:
Let's merge this one now and discuss these changes in depth on the PR for the template repo.